### PR TITLE
Add crude implementation

### PR DESCRIPTION
**Pull request by Hacksore**
@yushijinhun I'd like to have this PR open as a discussion for my learning on how `loadSkinToCanvas` would work for us in the 2d sense.

Thanks!

### CODE REVIEW

File: src/skinview2d.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// import { loadSkinToCanvas } from "skinview-utils";
+
+class SkinvViewer {
```
Comment (yushijinhun) [inline src/skinview2d.ts]: Maybe it's better to use `Skin2DViewer` to avoid conflicts?

File: src/skinview2d.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    this.skinImage = new Image();
+    this.skinImage.src = imagePath;
+
+    // texture loading
+    this.skinImage.crossOrigin = "anonymous";
+    this.skinImage.onerror = (): void => console.error("Failed loading " + this.skinImage.src);
+    this.skinImage.onload = (): void => {
+      // Not 100% sure I understand the benifit of this, right now it will just render the image
+      // directly to the canvas, which in the 2d skin viewer case is not what we want?
+      // loadSkinToCanvas(this.canvas, this.skinImage);
```
Comment (yushijinhun) [inline src/skinview2d.ts]: My idea is to split this project into two parts:
 * One part provides some low-level functions to draw the player, like `model.ts` in skinview3d, for example:
    ```typescript
    drawHeadFront(ctx: CanvasRenderingContext2D, skinTexture: TextureCanvas, x: number, y: number, w:number, h: number);
    ```
 * Another part is `Skin2DViewer`, which provides high-level interfaces, like `viewer.ts` in skinview3d.

Comment (yushijinhun) [reply src/skinview2d.ts]:
```typescript
this.skinImage.crossOrigin = "anonymous";
this.skinImage.onerror = (): void => console.error("Failed loading " + this.skinImage.src);
this.skinImage.onload = (): void => { ...
```
I think this piece of code is a bit redundant (same code is in skinview3d).

I'm trying to extract the common part to skinview-utils, but I'm not satisfied with my current solution (see https://github.com/bs-community/skinview-utils/issues/3). Maybe it's a bit over design?

File: src/skinview2d.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    }
+    const { skinImage, ctx } = this;
+    const { width, height } = this.canvas;
+
+    this.ctx.imageSmoothingEnabled = false;
+
+    // test drawing head with hard coded values for now
+    ctx.drawImage(skinImage, 40, 8, 8, 8, 0, 0, (width/8) * 8, (height/8) * 8);
+
+    window.requestAnimationFrame(() => this.render());
```
Comment (yushijinhun) [inline src/skinview2d.ts]: Do we really need animations?

**Review by yushijinhun — COMMENTED** [review]
It seems that I forgot `.editorconfig` so there is now a spaces-vs-tabs fight. I'll add it.

**Comment by yushijinhun** [comment]
What `loadSkinToCanvas` does is convert a (maybe old-fashioned) skin texture to modern format. And then we use the "standardized" skin texture to render things.

File: src/skinview2d.ts (hunk 1 above)
Comment (Hacksore) [reply src/skinview2d.ts]: what about `SkinViewer2D`?

File: src/skinview2d.ts (hunk 3 above)
Comment (Hacksore) [reply src/skinview2d.ts]: Unlikely, I will refactor this later.

File: package.json [hunk 4]
```diff
@@ -1,45 +1,50 @@
...
+    "skinview-utils": "^0.2.1"
+  },
+  "devDependencies": {
+    "@babel/core": "^7.8.4",
+    "@rollup/plugin-node-resolve": "^7.0.0",
+    "@rollup/plugin-typescript": "^3.0.0",
+    "@storybook/html": "^5.3.9",
+    "@typescript-eslint/eslint-plugin": "^2.18.0",
+    "@typescript-eslint/parser": "^2.18.0",
+    "babel-loader": "^8.0.6",
```
Comment (g-plane) [inline package.json]: What does require Babel?

File: src/viewer.ts [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+// TODO: create config for class
+
+enum RenderPart {
+	HEAD_FRONT = 0,
+	HEAD_BACK = 1,
+	BODY_FRONT = 2,
+	BODY_BACK = 3
+}
+
+interface SkinvViewer2DOptions {
```
Comment (g-plane) [inline src/viewer.ts]: Typo?

File: src/viewer.ts [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+	domElement: HTMLElement;
+	skinUrl: string;
+	width?: number;
+	height?: number;
+	toRender?: string;
+}
+
+class SkinvViewer2D {
```
Comment (g-plane) [inline src/viewer.ts]: Typo?

**Review by g-plane — COMMENTED** [review]
"Skinv" -> "Skin"?

File: package.json (hunk 4 above)
Comment (Hacksore) [reply package.json]: Storybook uses them.

File: package.json [hunk 7]
```diff
@@ -29,19 +29,23 @@
   },
   "homepage": "https://github.com/bs-community/skinview2d#readme",
   "dependencies": {
+    "babel-loader": "^8.0.6",
```
Comment (g-plane) [inline package.json]: Is this intended?

Comment (Hacksore) [reply package.json]: Yeah sorry. I already resolved the last one but did not make a comment until after, it's for storybook.

Comment (Hacksore) [reply package.json]: I need to move it back to devDeps.

**Pull request merged**